src/MizarWidgets/include/MizarWidgets/SamplingWithFrameTrackReportModel.h

```diff
@@ -0,0 +1,170 @@
+class SamplingWithFrameTrackReportModelTmpl : public QAbstractTableModel {
+  template <typename T>
+  using Baseline = ::orbit_mizar_base::Baseline<T>;
+  template <typename T>
+  using Comparison = ::orbit_mizar_base::Comparison<T>;
+  using SFID = ::orbit_mizar_base::SFID;
+
+ public:
+  enum class Column {
+    kFunctionName,
+    kBaselineExclusivePercent,
+    kBaselineExclusiveTimePerFrame,
+    kComparisonExclusivePercent,
+    kComparisonExclusiveTimePerFrame,
+    kPvalue,
+    kIsSignificant,
+    kSlowdownPercent,
+    kSlowdownPerFrame
+  };
+
+  static constexpr int kColumnsCount = 9;
+
+  explicit SamplingWithFrameTrackReportModelTmpl(Report report, QObject* parent = nullptr)
+      : QAbstractTableModel(parent), report_(std::move(report)) {
+    for (const auto& [sfid, unused_name] : report_.GetSfidToNames()) {
+      if (*BaselineExclusiveCount(sfid) > 0 || *ComparisonExclusiveCount(sfid) > 0) {
+        sfids_.push_back(sfid);
+      }
+    }
+  }
+
+  [[nodiscard]] int rowCount(const QModelIndex& /*parent*/) const override {
+    return static_cast<int>(sfids_.size());
+  };
+  [[nodiscard]] int columnCount(const QModelIndex& /*parent*/) const override {
+    return kColumnsCount;
+  };
+  [[nodiscard]] QVariant data(const QModelIndex& index, int role) const override {
+    if (role != Qt::DisplayRole) return {};
+    return QString::fromStdString(MakeDisplayedString(index));
+  }
+
+  [[nodiscard]] QVariant headerData(int section, Qt::Orientation orientation,
+                                    int role) const override {
+    static const absl::flat_hash_map<Column, QString> kColumnNames = {
+        {Column::kFunctionName, "Function"},
+        {Column::kBaselineExclusivePercent, "Baseline, %"},
+        {Column::kBaselineExclusiveTimePerFrame, "Baseline (per frame), us"},
+        {Column::kComparisonExclusivePercent, "Comparison, %"},
+        {Column::kComparisonExclusiveTimePerFrame, "Comparison (per frame), us"},
+        {Column::kPvalue, "P-value"},
+        {Column::kIsSignificant, "Significant?"},
+        {Column::kSlowdownPercent, "Slowdown, %"},
+        {Column::kSlowdownPerFrame, "Slowdown (per frame), us"}};
+
+    if (orientation == Qt::Vertical || role != Qt::DisplayRole) {
+      return {};
+    }
+
+    return kColumnNames.at(static_cast<Column>(section));
+  }
+
+ private:
+  [[nodiscard]] std::string MakeDisplayedString(const QModelIndex& index) const {
+    const SFID sfid = sfids_[index.row()];
+    const auto column = static_cast<Column>(index.column());
+    switch (column) {
+      case Column::kFunctionName:
+        return report_.GetSfidToNames().at(sfid);
+      case Column::kBaselineExclusivePercent:
+      case Column::kBaselineExclusiveTimePerFrame:
+      case Column::kComparisonExclusivePercent:
+      case Column::kComparisonExclusiveTimePerFrame:
+        return absl::StrFormat("%.3f", (MakeNumericEntry(sfid, column)));
```

Comment:
What's the rationale for having `MakeNumericEntry`? Is it to avoid having multiple `absl::StrFormat` calls?

Reply:
If that was the only reason I would argue that it would be more readable to have redundant StrFormat calls over the separate function - mainly because the Column enum makes it hard to distinguish which cases need to be handled in which function and it ties the two function together anyway.